Adds Direct VPC egress support for Cloud Run V2 resources

yanweiguo opened the pull request:
<!-- Put a description of what this PR is for here, along with any references to issues that this resolves or contributes to -->
Fixes https://github.com/hashicorp/terraform-provider-google/issues/15568

<!--
Please self-review your PR against the review checklist before creating it: https://googlecloudplatform.github.io/magic-modules/contribute/review-pr/

Completing the checklist will help speed up the review process, and we appreciate you spending time on them before sending
your code to be reviewed.

If your PR is still work in progress, please create it in draft mode
-->

<!-- AUTOCHANGELOG for Downstream PRs.

Please select one of the following "release-note:" headings:
    - release-note:enhancement
    - release-note:bug
    - release-note:note
    - release-note:new-resource
    - release-note:new-datasource
    - release-note:deprecation
    - release-note:breaking-change
    - release-note:none

Unless you choose release-note:none, please add a release note.

See https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/ for writing good release notes.

You can add more release note blocks if you want more than one CHANGELOG
entry for this PR.
-->
**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
cloudrunv2: added fields `network_interfaces` to resource `google_cloud_run_v2_job` to support Direct VPC egress.
```

```release-note:enhancement
cloudrunv2: added fields `network_interfaces` to resource `google_cloud_run_v2_service` to support Direct VPC egress.
```

modular-magician commented:
Hello! I am a robot. It looks like you are a: ~Community Contributor~ Googler ~Core Contributor~. Tests will run automatically.

@shuyama1, a repository maintainer, has been assigned to review your changes. If you have not received review feedback within 2 business days, please leave a comment on this PR asking them to take a look.

You can help make sure that review is quick by [doing a self-review](https://googlecloudplatform.github.io/magic-modules/contribute/review-pr/) and by [running impacted tests locally](https://googlecloudplatform.github.io/magic-modules/get-started/run-provider-tests/).

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-8932-old..auto-pr-8932) ( 6 files changed, 456 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-8932-old..auto-pr-8932) ( 6 files changed, 456 insertions(+))
TF Conversion: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-8932-old..auto-pr-8932) ( 2 files changed, 110 insertions(+))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-8932-old..auto-pr-8932) ( 8 files changed, 244 insertions(+))
## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_cloud_run_v2_service` (25 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_cloud_run_v2_service" "primary" {
  template {
    vpc_access {
      network_interfaces {
        subnetwork = # value needed
        tags       = # value needed
      }
    }
  }
}

```

modular-magician commented:
#### Tests analytics 
Total tests: `3039` 
Passed tests `2736` 
Skipped tests: `297` 
Affected tests: `6` 

#### Action taken 
<details> <summary>Found 6 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccCloudRunV2Job_cloudrunv2JobDirectvpcExample|TestAccCloudRunV2Service_cloudrunv2ServiceDirectvpcExample|TestAccGKEHub2Scope_gkehubScopeBasicExample|TestAccGKEHub2ScopeIamMemberGenerated|TestAccGKEHub2ScopeIamBindingGenerated|TestAccGKEHub2ScopeIamPolicyGenerated </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-8932-old..auto-pr-8932) ( 6 files changed, 460 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-8932-old..auto-pr-8932) ( 6 files changed, 460 insertions(+))
TF Conversion: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-8932-old..auto-pr-8932) ( 2 files changed, 110 insertions(+))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-8932-old..auto-pr-8932) ( 8 files changed, 246 insertions(+))

modular-magician commented:
#### Tests analytics 
Total tests: `3039` 
Passed tests `2733` 
Skipped tests: `297` 
Affected tests: `9` 

#### Action taken 
<details> <summary>Found 9 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccBigtableAppProfile_bigtableAppProfileAnyclusterExample|TestAccBigtableAppProfile_bigtableAppProfileSingleclusterExample|TestAccBigtableAppProfile_bigtableAppProfileMulticlusterExample|TestAccCloudRunV2Service_cloudrunv2ServiceDirectvpcExample|TestAccCloudRunV2Job_cloudrunv2JobDirectvpcExample|TestAccGKEHub2Scope_gkehubScopeBasicExample|TestAccGKEHub2ScopeIamBindingGenerated|TestAccGKEHub2ScopeIamMemberGenerated|TestAccGKEHub2ScopeIamPolicyGenerated </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$ 
 `TestAccBigtableAppProfile_bigtableAppProfileAnyclusterExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8932/artifacts/7c445d15-6c43-4d43-a071-9aae73f7ae6c/recording/TestAccBigtableAppProfile_bigtableAppProfileAnyclusterExample.log)]
`TestAccBigtableAppProfile_bigtableAppProfileSingleclusterExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8932/artifacts/7c445d15-6c43-4d43-a071-9aae73f7ae6c/recording/TestAccBigtableAppProfile_bigtableAppProfileSingleclusterExample.log)]
`TestAccBigtableAppProfile_bigtableAppProfileMulticlusterExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8932/artifacts/7c445d15-6c43-4d43-a071-9aae73f7ae6c/recording/TestAccBigtableAppProfile_bigtableAppProfileMulticlusterExample.log)]
`TestAccCloudRunV2Service_cloudrunv2ServiceDirectvpcExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8932/artifacts/7c445d15-6c43-4d43-a071-9aae73f7ae6c/recording/TestAccCloudRunV2Service_cloudrunv2ServiceDirectvpcExample.log)]
`TestAccCloudRunV2Job_cloudrunv2JobDirectvpcExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8932/artifacts/7c445d15-6c43-4d43-a071-9aae73f7ae6c/recording/TestAccCloudRunV2Job_cloudrunv2JobDirectvpcExample.log)]
`TestAccGKEHub2Scope_gkehubScopeBasicExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8932/artifacts/7c445d15-6c43-4d43-a071-9aae73f7ae6c/recording/TestAccGKEHub2Scope_gkehubScopeBasicExample.log)]
`TestAccGKEHub2ScopeIamBindingGenerated`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8932/artifacts/7c445d15-6c43-4d43-a071-9aae73f7ae6c/recording/TestAccGKEHub2ScopeIamBindingGenerated.log)]
`TestAccGKEHub2ScopeIamMemberGenerated`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8932/artifacts/7c445d15-6c43-4d43-a071-9aae73f7ae6c/recording/TestAccGKEHub2ScopeIamMemberGenerated.log)]
`TestAccGKEHub2ScopeIamPolicyGenerated`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8932/artifacts/7c445d15-6c43-4d43-a071-9aae73f7ae6c/recording/TestAccGKEHub2ScopeIamPolicyGenerated.log)] 

##### Rerun these tests in REPLAYING mode to catch issues 

$\textcolor{green}{\textsf{No issues found for passed tests after REPLAYING rerun.}}$ 

---
$\textcolor{green}{\textsf{All tests passed!}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8932/artifacts/7c445d15-6c43-4d43-a071-9aae73f7ae6c/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-8932/artifacts/7c445d15-6c43-4d43-a071-9aae73f7ae6c/recording) for each test

shuyama1 reviewed (commented):
Are any of these fields support in-place update? If so, would you mind adding/ modifying an update test to the update functionality of it?

yanweiguo commented on mmv1/templates/terraform/examples/cloudrunv2_job_directvpc.tf.erb:
```diff
@@ -0,0 +1,26 @@
+        network_interfaces {
+          network = "default"
+          subnetwork = "default"
+          tags = ["tag1"]
```
Done

yanweiguo commented:
> Are any of these fields support in-place update? If so, would you mind adding/ modifying an update test to the update functionality of it?

Added tests. For Service, I can only use existing "default" network. For Job, non-existing networks are fine.

While adding the tests, found that existing `egress` field lacks default setting. Added `default_from_api: true` since our API set a default value for this field.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-8932-old..auto-pr-8932) ( 8 files changed, 632 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-8932-old..auto-pr-8932) ( 8 files changed, 632 insertions(+))
TF Conversion: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-8932-old..auto-pr-8932) ( 2 files changed, 110 insertions(+))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-8932-old..auto-pr-8932) ( 8 files changed, 246 insertions(+))

modular-magician commented:
#### Tests analytics 
Total tests: `3049` 
Passed tests `2747` 
Skipped tests: `298` 
Affected tests: `4` 

#### Action taken 
<details> <summary>Found 4 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccCloudRunV2Job_cloudrunv2JobWithDirectVPCUpdate|TestAccCloudRunV2Service_cloudrunv2ServiceWithDirectVPCUpdate|TestAccCloudRunV2Service_cloudrunv2ServiceDirectvpcExample|TestAccCloudRunV2Job_cloudrunv2JobDirectvpcExample </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$ 
 `TestAccCloudRunV2Job_cloudrunv2JobWithDirectVPCUpdate`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8932/artifacts/54eaf0a3-4339-487b-b901-199bd3df297f/recording/TestAccCloudRunV2Job_cloudrunv2JobWithDirectVPCUpdate.log)]
`TestAccCloudRunV2Service_cloudrunv2ServiceWithDirectVPCUpdate`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8932/artifacts/54eaf0a3-4339-487b-b901-199bd3df297f/recording/TestAccCloudRunV2Service_cloudrunv2ServiceWithDirectVPCUpdate.log)]
`TestAccCloudRunV2Service_cloudrunv2ServiceDirectvpcExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8932/artifacts/54eaf0a3-4339-487b-b901-199bd3df297f/recording/TestAccCloudRunV2Service_cloudrunv2ServiceDirectvpcExample.log)]
`TestAccCloudRunV2Job_cloudrunv2JobDirectvpcExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8932/artifacts/54eaf0a3-4339-487b-b901-199bd3df297f/recording/TestAccCloudRunV2Job_cloudrunv2JobDirectvpcExample.log)] 

##### Rerun these tests in REPLAYING mode to catch issues 

$\textcolor{green}{\textsf{No issues found for passed tests after REPLAYING rerun.}}$ 

---
$\textcolor{green}{\textsf{All tests passed!}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8932/artifacts/54eaf0a3-4339-487b-b901-199bd3df297f/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-8932/artifacts/54eaf0a3-4339-487b-b901-199bd3df297f/recording) for each test

shuyama1 reviewed (approved):
Thank you!
